crates/era/src/era1_types.rs

```diff
@@ -0,0 +1,219 @@
+        starting_number_bytes.copy_from_slice(&entry.data[0..8]);
+        let starting_number = u64::from_le_bytes(starting_number_bytes);
+
+        // Extract count = last 8 bytes
+        let mut count_bytes = [0u8; 8];
+        count_bytes.copy_from_slice(&entry.data[entry.data.len() - 8..]);
+        let count = u64::from_le_bytes(count_bytes) as usize;
+
+        // Verify that the entry has the correct size
+        let expected_size = 8 + count * 8 + 8;
+        if entry.data.len() != expected_size {
+            return Err(E2sError::Ssz(format!(
+                "SlotIndex entry has incorrect size: expected {}, got {}",
+                expected_size,
+                entry.data.len()
+            )));
+        }
+
+        // Extract all offsets
+        let mut offsets = Vec::with_capacity(count);
+        for i in 0..count {
+            let start = 8 + i * 8;
+            let end = start + 8;
+            let mut offset_bytes = [0u8; 8];
+            offset_bytes.copy_from_slice(&entry.data[start..end]);
+            offsets.push(i64::from_le_bytes(offset_bytes));
+        }
+
+        Ok(Self { starting_number, offsets })
+    }
+}
+
+/// Era1 file identifier
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Era1Id {
+    /// Network configuration name
+    pub config_name: String,
+
+    /// First block number in file
+    pub start_block: BlockNumber,
+
+    /// Number of blocks in the file
+    pub block_count: u32,
+
+    /// Chain identifier
+    pub chain_id: ChainId,
+}
+
+impl Era1Id {
+    /// Create a new [`Era1Id`]
+    pub fn new(
+        config_name: impl Into<String>,
+        start_block: BlockNumber,
+        block_count: u32,
+        chain_id: ChainId,
+    ) -> Self {
+        Self { config_name: config_name.into(), start_block, block_count, chain_id }
+    }
+
+    /// Convert to file name following the era1 file naming:
+    /// `<config-name>-<start-block>-<block-count>-<chain-id>.era1`
+    /// inspired from era file naming convention in
+    /// <https://github.com/eth-clients/e2store-format-specs/blob/main/formats/era.md#file-name>
+    /// See also <https://github.com/eth-clients/e2store-format-specs/blob/main/formats/era1.md>
+    pub fn to_file_name(&self) -> String {
+        format!(
+            "{}-{:10}-{:02}-{}.era1",
+            self.config_name, self.start_block, self.block_count, self.chain_id
+        )
+    }
+}
+
+/// Group in an Era1 file
+///
+/// Format: `Version | block-tuple* | other-entries* | Accumulator | BlockIndex`
+#[derive(Debug)]
+pub struct Era1Group {
+    /// Blocks in this era1 group
+    pub blocks: Vec<BlockTuple>,
+
+    /// Other entries that don't fit into the standard categories
+    pub other_entries: Vec<Entry>,
+
+    /// Accumulator is hash tree root of block headers and difficulties
+    pub accumulator: Accumulator,
+
+    /// Block index, optional, omitted for genesis era
+    pub block_index: BlockIndex,
+}
+
+impl Era1Group {
```

Comment:
I think here we also want a helper type that reads this from a given file for convenience